test/integration/CommunityEndpoints.test.js

```diff
@@ -0,0 +1,246 @@
+import assert from 'assert'
+
+import { Contract, providers, utils, Wallet } from 'ethers'
+
+import StreamrClient from '../../src'
+import * as Token from '../../contracts/TestToken.json'
+
+import config from './config'
+
+function sleep(ms) {
+    return new Promise((resolve) => {
+        setTimeout(resolve, ms)
+    })
+}
+
+describe('CommunityEndPoints', () => {
+    let community
+
+    let testProvider
+    let adminClient
+    let adminWallet
+
+    beforeAll(async () => {
+        testProvider = new providers.JsonRpcProvider(config.ethereumServerUrl)
+        const network = await testProvider.getNetwork().catch((e) => {
+            throw new Error(`Connecting to Ethereum failed, config = ${JSON.stringify(config)}`, e)
+        })
+        console.log('Connected to Ethereum network: ', JSON.stringify(network))
+
+        adminWallet = new Wallet(config.privateKey, testProvider)
+        adminClient = new StreamrClient({
+            auth: {
+                privateKey: adminWallet.privateKey
+            },
+            autoConnect: false,
+            autoDisconnect: false,
+            ...config.clientOptions,
+        })
+
+        console.log('beforeAll done')
+    })
+    beforeEach(async () => {
+        console.log('starting beforeEach')
+        await adminClient.ensureConnected()
+        console.log('deploying new community...')
+        community = await adminClient.deployCommunity({
+            provider: testProvider,
+        })
+        console.log(`Going to deploy to ${community.address}`)
+        await community.deployed()
+        console.log(`Deployment done for ${community.address}`)
+        await community.isReady(2000, 200000)
+        console.log(`Community ${community.address} is ready to roll`)
+        await adminClient.createSecret(community.address, 'secret', 'CommunityEndpoints test secret')
+    }, 300000)
```

Comment:
is there a way we can get this to take less than 5 minutes? This is a very long time.

If not, I wonder if we should move such glacially slow tests into a different directory so everything else can be tested easily.

Reply:
I made another ticket of trimming that down https://streamr.atlassian.net/browse/CPS-85

I don't think it actually will take 5 minutes, but Travis is slow sometimes...